test/knowledge_panels_from_json_test.dart

```diff
@@ -422,4 +427,67 @@ void main() {
     KnowledgePanels kp = KnowledgePanels.fromJson(panels);
     expect(kp.panelIdToPanelMap.length, equals(8));
   });
+
+  // Ensure that a "text_element" with some sources data is well parsed
+  test('Load knowledge panel JSON with a text element', () {
+    final String id = 'additive_en:e262';
+
+    final Map<String, dynamic> jsonData = <String, dynamic>{
```

Comment:
Given that `'additive_en:e262'` is not in your json string I would be surprised if the test actually passed.